### ISSUE TITLE
Terrain-Masked Radio Propagation

### ISSUE THREAD

**Issue by MBarkerUK** [lErrorl404l/AEE#32]
## Real Propagation Physics Added (2026-09-16)

Terrain-masked propagation (#32) and evaporation ducting (#37) now have grounded physics from ITU-R P.526, P.530, and the classic duct literature.

### Knife-edge diffraction (ITU-R P.526-16 section 4.1)
```
nu = h*sqrt(2(d1+d2)/(lambda*d1*d2))     // h = excess height above Tx-Rx line
J(nu) = 6.9 + 20*log10(sqrt((nu-0.1)^2+1) + nu - 0.1) dB   // valid nu > -0.78
```
Key values: nu=-0.78 -> 0 dB; nu=0 -> 6.0 dB; nu=1 -> 13.7 dB; nu=2 -> 21.9 dB.

### Method choice (cost vs accuracy)
- **Bullington**: O(N), two passes + one J(nu). CHEAPEST. Least accurate (NTIA TR-26-580). Use per tick, escalate to Deygout on marginal links.
- **Deygout**: main edge = max nu, recurse on sub-paths, max 3-4 edges. Most accurate, O(2N).
- **Epstein-Peterson**: L = sum J(nu_i), one pass, overestimates close edges.

### Bullington algorithm (P.452-18 section 4.2.3)
```
S_te = max_i (h_i - h_ts + 500*C_e*d_i*(d_p-d_i))/d_i
S_re = max_i (h_i - h_rs + 500*C_e*d_i*(d_p-d_i))/(d_p-d_i)
S_tr = (h_rs-h_ts)/d_p
if S_te < S_tr:  // LOS
    nu_max over all points; loss = J(nu_max)
else:
    d_b = (h_rs-h_ts+S_re*d_p)/(S_te+S_re)
    h_b = h_ts+S_te*d_b-(h_ts*(d_p-d_b)+h_rs*d_b)/d_p
    nu_b = |h_b|*sqrt(0.002*d_p/(lambda*d_b*(d_p-d_b)))
    loss = J(nu_b)
C_e = 1/8500 per km (k=4/3)
```

### Fresnel zone clearance (P.530-19 section 2.2.2)
```
F1 = sqrt(lambda*d1*d2/(d1+d2))
```
Clear at c >= 0.6*F1 (nu = -0.85, J = -0.45 dB). Planning: 1.0 F1 temperate, 0.6 F1 tropical >30 km, 0.3 F1 extended obstruction, 0.0 F1 single isolated obstacle.

### Computational cost (honest)
- P.1812 sampling 30-50 m: 10 km -> 200 samples; 50 km -> 1000 samples
- Stretched-string analysis (P.526 section 4.4) reduces to 1-5 edges
- 10-20 links x 300 samples x 5 us = 15 ms per tick — TOO MUCH at 60 Hz
- **Fix**: cache terrain profiles, recompute at 1 Hz or on 50 m movement, sample at 50-100 m. Math on reduced edges < 1 ms.

### Physics
```
M = N + 0.157*z;  N = 77.6*p/T + 3.73e5*e/T^2
```
Duct when dM/dz < 0 (gradient below -0.157 N-units/m).

Paulus-Jeske: M(z) = M0 + 0.125*[z - delta*ln((z+z0)/z0)], z0 = 1.5e-4 m.

### Duct height delta
World avg 13 m, tropics to 40 m, Gulf 20 m, North Sea 5-6 m. Stable air-sea temperature difference raises delta.

### Cutoff frequency (Hall, validated)
```
lambda_max = 2.5e-3*sqrt(deltaN/H - 0.157)*H^(3/2)
f_min = c/lambda_max
```
Example: H=13, deltaN=10 -> f_min ~ 3.3 GHz. Duct works above 1-2 GHz, best 10-18 GHz. (Note: VHF/UHF below ~1 GHz is largely NOT ducted — the duct mainly affects SHF. Honest finding: the original VHF ducting assumption was wrong for the frequency band; the evaporation duct is an SHF phenomenon.)

### Surface vs elevated duct
- Evaporation duct: surface-based, weak OTH (~44.8 km avg)
- Elevated ducts: strong OTH (94-106 km)
- Turton constant: 3.77e-3 surface; elevated disputed

### Validation numbers
- Ducting exceeds free-space over 500+ km sea (P.452)
- Huang 2022: 27.8 m duct -> 50 km OTH; 6.8 m -> none
- Shi 2019: 149 km path, cutoff 2.6 GHz
- Hitney & Veith: 17.7 GHz, 81 km
- Range extension 3-10x (13 km horizon -> 45-150 km ducted)
- Cap ducted range 50-150 km, attenuation 0.1-0.5 dB/km

## ITU-R Model Mapping

| Rec | Covers |
|---|---|
| P.526 | Diffraction (knife-edge, multiple edge) |
| P.530 | Terrestrial LOS clearance |
| P.618 | Earth-space (rain, scintillation) |
| P.2001 | Ducting statistics for interference |
| P.452 | Interference/ducting model (Lba, beta0) |
| P.1812 | Point-to-area |

## Honest Correction

The evaporation duct is an **SHF phenomenon** (1-40 GHz, best 10-18 GHz). VHF/UHF military radios (30-300 MHz) are largely unaffected. The ducting in #37 should apply to radar/satcom SHF links, not VHF tactical radios. The existing radio addon's VHF ducting model may need re-scoping.

## Test Vectors

- Single ridge nu=18.97, J~32.5 dB
- Grazing: J=6.0 dB
- Evap duct: H=13, deltaN=10 -> f_min ~ 3.3 GHz

## Sources

- ITU-R P.526-16 (2025-11) knife-edge diffraction
- ITU-R P.530-19 clearance criteria
- ITU-R P.452-18 (Bullington algorithm)
- NTIA TR-26-580 (method comparison)
- Paulus & Jeske (1981) evaporation duct
- Hall (2003?) duct cutoff validation
- Huang 2022, Shi 2019, Hitney & Veith (range extension)
